### ISSUE TITLE
Python shenanigans (data.get or compatibility)

### ISSUE THREAD

**Issue by JabLuszko** [pokealarm/pokealarm#691]
## Troubleshooting Checklist

- [x] My Scanner install is at the most recent version.
- [x] My PokeAlarm install is at the most recent version.
- [x] I've already checked the
 [FAQ](https://github.com/RocketMap/PokeAlarm/wiki/faq) and the
 [Wiki](https://github.com/RocketMap/PokeAlarm/wiki) already.

## Problem Description
This was debugged inside WeatherEvent, but should impact more things that use diff fields for MAD/RDM.
Initial error was: ```ValueError: '=' alignment not allowed in string format specifier```
Turns out severity_id was None, while it was sent in webhook as "alert_severity" - yet "severity" (probably used by RDM) was not set, so was returning None.

```
self.severity_id = data.get('alert_severity') or data.get('severity')

Log:
data.get('alert_severity'): 0
data.get('severity'): None

>>> print 0 or None
None
>>> print None or 0
0

Python3 does the same thing.
```

## Environment
Operating System: ` Linux `
Python Version: ` Python 2.7.15+ (default, Nov 27 2018, 23:36:35)
[GCC 7.3.0] on linux2`

**Comment by LegitDongo** [comment]
`alert_severity` was getting sent as `0` - a valid value
I'm assuming that Python evaluates this as "falsey", thus the `or` skips over it

**Comment by LegitDongo** [comment]
https://docs.python.org/2/reference/expressions.html#boolean-operations
Documentation reference
```
The expression x and y first evaluates x; if x is false, its value is returned; otherwise, y is evaluated and the resulting value is returned.

The expression x or y first evaluates x; if x is true, its value is returned; otherwise, y is evaluated and the resulting value is returned.
```

**Comment by JimBim89** [comment]
Line 67 may also be a culprit here, changing "is" to "==" should be the way to go.

(WeatherEvent.py:67:23: F632 use ==/!= to compare str, bytes, and int literals)

**Comment by JimBim89** [comment]
Also when weather condition is 0 (ingame weather provider down) it will be displayed as unknown in PA alerts instead of using the available locale "None" for id 0.

**Comment by LegitDongo** [comment]
The problem is that lazy evaluation treats `0` as `falsey`, so my plan was to change it to something like
`data.get('alert_severity', data.get('severity', 0))` in order to only go to the next `data.get` on `None`
e.g. 
`data.get('alert_severity') or data.get('severity')` equates to `0 or None`, which is why `weather.severity` is showing `None`

Also, I've gotten started on this; it will have to be changed across the board.

**Comment by LegitDongo** [comment]
#693 will be merged and has a weather fix, but doesn't fix the underlying issue.

**Comment by billyjbryant** [comment]
@LegitDongo is this still an issue that needs further research?

**Comment by LegitDongo** [comment]
@billyjbryant yes, this is still an issue; the impact is minimal because most of the scanners use the same set of keys
